### PR TITLE
Adding API authentication

### PR DESCRIPTION
**Pull request by Bhavesh-Patel**
Note: changes to [dotnet.yml workflow](https://github.com/City-of-Lincoln-Council/housing-repairs-online-api/commit/c4353e7b4d538fd6c3f482e33b95be347d4ad7e0) will be reverted prior to merge

### CODE REVIEW

**Comment by Bhavesh-Patel** [comment]
[Add authentication to Repairs API](https://trello.com/c/Gn6108nd/194-add-authentication-to-repairs-api)

File: .github/workflows/dotnet.yml [hunk 1]
```diff
@@ -36,7 +36,7 @@ jobs:
       - name: Test
         run: dotnet test
   build:
-    if: github.event_name == 'push' && github.ref == 'refs/heads/main'
+    #if: github.event_name == 'push' && github.ref == 'refs/heads/main'
```
Comment (elena-vi) [inline .github/workflows/dotnet.yml:39]: probably want to restore these

File: .github/workflows/dotnet.yml [hunk 2]
```diff
@@ -61,7 +61,7 @@ jobs:
           path: ${{env.DOTNET_ROOT}}/myapp
   deploy:
     name: Deploy
-    if: github.event_name == 'push' && github.ref == 'refs/heads/main'
+    #if: github.event_name == 'push' && github.ref == 'refs/heads/main'
```
Comment (elena-vi) [inline .github/workflows/dotnet.yml:64]:
```suggestion
    if: github.event_name == 'push' && github.ref == 'refs/heads/main'
```

**Review by elena-vi — COMMENTED** [review]
Some documentation about how requests should be made to the API would be good

File: .github/workflows/dotnet.yml (hunk 1 above)
Comment (Bhavesh-Patel) [reply .github/workflows/dotnet.yml]: Restored ✅

**Comment by Bhavesh-Patel** [comment]
> Some documentation about how requests should be made to the API would be good

Added documentation - see README.md

**Review by elena-vi — APPROVED** [review]

**Pull request merged**